src/App.tsx

```diff
@@ -1,19 +1,169 @@
 import React from 'react';
-import './App.css';
+import './App.scss';
 
+import todos from './api/todos';
 import users from './api/users';
+import { TodoList } from './components/TodoList/TodoList';
+import { Todo } from './react-app-env';
 
-const App: React.FC = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
-};
+const preparedTodos: Todo[] = todos.map(todo => {
+  return ({
+    user: users.find(user => user.id === todo.userId) || null,
+    title: todo.title,
+    completed: todo.completed,
+    id: todo.id,
+  });
+});
+
+interface State {
+  userName: string;
+  title: string;
+  stateTodos: Todo[];
+  titleIsValid: boolean;
+  userNameIsValid: boolean;
+}
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    userName: '',
+    title: '',
+    stateTodos: preparedTodos,
+    titleIsValid: true,
+    userNameIsValid: true,
+  };
+
+  clearForm = () => {
+    this.setState({
+      userName: '',
+      title: '',
+    });
+  };
+
+  addUser = (event: React.FormEvent<HTMLFormElement>) => {
+    const { title, userName } = this.state;
+
+    event.preventDefault();
+
+    this.setState((currentState) => ({
+      titleIsValid: !!currentState.title,
+      userNameIsValid: !!currentState.userName,
+    }));
+
+    if (!!userName && !!title) {
+      this.setState((currentState) => {
+        const temporaryTodos = [...currentState.stateTodos];
+        const maxId = temporaryTodos
+          .reduce((prev, curr) => (prev.id > curr.id ? prev : curr)).id + 1;
+
+        temporaryTodos.push({
+          user: users.find((user) => user.name === userName) || null,
+          title,
+          completed: false,
+          id: maxId,
+        });
+
+        return ({
+          stateTodos: temporaryTodos,
+        });
+      }, this.clearForm);
+    }
+  };
+
+  handleChangeSelect = (event: React.FormEvent<HTMLSelectElement>) => {
+    const { value } = event.currentTarget;
+
+    this.setState({
+      userName: value,
+      userNameIsValid: !!value,
```

Comment:
it shouldn't show error if the user started to type, then have cleared input but don't press Add todo